### PR TITLE
(A lot of) Content for a SLE12 STIG profile

### PR DESCRIPTION
**Pull request by maltek**
Hi,

we've been working in our own fork on rules for the SLE12 STIG. This is a huge diff, which contains everything - simple changes to enable rules also on SLE12, some new if-conditions when SLE is a bit different, bug fixes for some rules, additional OVAL checks and BASH remediations for rules missing them, some completely new rules, a few changes to the generation scripts, ...

With that size, I don't expect you to just merge this as-is, but hopefully we can get there over time.

### CODE REVIEW

**Comment by jan-cerny** [comment]
@openscap-ci ok to test

File: shared/checks/oval/installed_env_has_yum_or_zypper_package.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <metadata>
+      <title>Package yum or zypper is installed</title>
+      <affected family="unix">
+        <platform>multi_platform_all</platform>
+      </affected>
+      <description>Checks if package yum or zypper is installed.</description>
+      <reference ref_id="cpe:/a:yum_or_zypper" source="CPE" />
+    </metadata>
+    <criteria>
+      <extend_definition comment="Package yum is installed" definition_ref="test_env_has_yum_installed" />
```
Comment (yuumasato) [inline shared/checks/oval/installed_env_has_yum_or_zypper_package.xml]: Reference the definition, `installed_env_has_yum_package`.
This should fix the build issue.

**Comment by yuumasato** [comment]
Wow, this is huge, and looks pretty good in general. It will take some time to review this.
I personally would prefer to review smaller PRs, focused on groups of rules.

**Comment by redhatrises** [comment]
> Wow, this is huge, and looks pretty good in general. It will take some time to review this.
> I personally would prefer to review smaller PRs, focused on groups of rules.

Agree with @yuumasato due to the sheer amount of changes.

A couple of things to change:

*  `nist@sle12` shouldn't exist and be appended to existing `nist:` references with no duplicates of course. Same thing for `srg`.
* Remove any and all instantiations of `stigid@rhel8`
* There should be no Jinja templating around YAML keys.
* If there is a reference to multiple STIGIDs, the content should be broken out into separate rules.
* There should only be one severity (if possible), and any severity of unknown should be replaced with `medium`.

**Comment by shawndwells** [comment]
@maltek If you're interested in assigning SLES-specific CCEs, I'd be happy to introduce you to the team at NIST who can assign them to you. If interested EMail me directly:  shawn@redhat.com.

**Comment by maltek** [comment]
OK, I'll open separate pull requests for individual groups of rules. It'll take me a bit of time, to sort through merge conflicts again and incorporate @redhatrises' feedback.

> There should be no Jinja templating around YAML keys.
I don't quite understand what you mean here. Can you rephrase/give more details please?

> If there is a reference to multiple STIGIDs, the content should be broken out into separate rules.
If that's how it has to be, then I can do that. But for some of those rules, that'll mean creating an identical copy because the STIGs have so much overlap (unfortunately).

> There should only be one severity (if possible), and any severity of unknown should be replaced with medium.
I wanted the severity to match that in the official STIGs, and didn't see a way to get there except for the ifs. But if we end up with some mismatches in the severities between this project and those STIGs, I think I could live with that, too.

**Comment by redhatrises** [comment]
> OK, I'll open separate pull requests for individual groups of rules. It'll take me a bit of time, to sort through merge conflicts again and incorporate @redhatrises' feedback.
> 
> > There should be no Jinja templating around YAML keys.
> 
> I don't quite understand what you mean here. Can you rephrase/give more details please?

For example, the `severity` is a YAML key, so there should be no templating around the yaml keys, so:
```
{{% if product == "sle12" %}}
severity: medium
{{% else %}}
severity: unknown
{{% endif %}}
```
should be 
```
severity: {{% if product == "sle12" -%}}medium
{{%- else -%}}
unknown
{{% endif -%}}
```

> > If there is a reference to multiple STIGIDs, the content should be broken out into separate rules.
> 
> If that's how it has to be, then I can do that. But for some of those rules, that'll mean creating an identical copy because the STIGs have so much overlap (unfortunately).
> 
> > There should only be one severity (if possible), and any severity of unknown should be replaced with medium.
> 
> I wanted the severity to match that in the official STIGs, and didn't see a way to get there except for the ifs. But if we end up with some mismatches in the severities between this project and those STIGs, I think I could live with that, too.

There are actually no mismatches in this PR. When you see a `severity` of `unknown`, it means that a severity has not been assigned yet. So all instances of:
```
{{% if product == "sle12" %}}
severity: medium
{{% else %}}
severity: unknown
{{% endif %}}
```
need to be changed to:
```
severity: medium
```

Also when you are handling merge conflicts, please do a `git rebase -i master` from a project standpoint we don't want to add the following type of commits to our git history:

- 43dbdd4
- e39fded

**Comment by maltek** [comment]
I'm just about to create my first wave of pull requests for some individual directories, with your early feedback integrated. Expect a trickle of more pull requests as I'm working through the directories. (I'm also avoiding adding the rules to `stig-sle12-disa.profile` for now, because handling merge conflicts in that file all the time is just annoying.)

I guess the main purpose of this pull request (starting a conversation on how to proceed) has been achieved, so feel free to close this. Thanks for the great feedback so far!

**Comment by redhatrises** [comment]
Closing per request
